File: Data Structures & Algorithms/binary-search/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        // nums is sorted

        int l = 0;
        int r = nums.length - 1;
        int m;
        int num;

        while (l <= r) {
            // m = (l + r) / 2;
            m = l + ((r - l) / 2);
            num = nums[m];
            if (num > target) {
                r = m - 1;
            } else if (num < target) {
                l = m + 1;
            } else {
                return m;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        ## 1. nested for loop, check nums[i] + nums[j] == target

        size  = len(nums)

        for i in range(size):
            for j in range (i+1, size):
                if nums[i] + nums[j] == target:
                    return [i, j]
        retunr [0,0]